test/features/caseList.feature

```diff
@@ -5,6 +5,10 @@ Feature: Set of Scenarios testing the functionality of the Case List table
     Given I have logged in
 
   Scenario: A Case Reference column in case list results has hyphens every 4 digits
+    Given the following definition for 'case fields'
```

Comment:
not sure I understand the value of having this step in this particular scenario. The scenario is not testing anything related to an optional field

Reply:
One more thing I've noticed in this PR, is that we added the new 'give the following definition step' to scenarios written in the old style. In my mind, we would have used this step only in scenarios written in the 'new style', i.e. the ones where all the fields mentioned in the 'given definition step' are then actually explicitly mentioned in the rest of the scenario. For example the scenario written in https://github.com/hmcts/ccd-case-management-web/pull/710/files. 
Maybe this is a good idea. The only concern I have is that it might result a bit confusing for who reads the scenario, because it doesn't understand why he has to pay attention to all the fields in the 'definition step' if they are then apparently not used in the scenario (but we know they are used under the hood).
I wanted to mention this to stimulate a discussion

Reply:
> One more thing I've noticed in this PR, is that we added the new 'give the following definition step' to scenarios written purely in the old style. For example the ones in `checkYourAnswers.feature`.
> In my mind, we would have used this step only in scenarios written in the 'new style', i.e. the ones where all the fields mentioned in the 'given definition step' are then actually _explicitly_ mentioned in the rest of the scenario. For example the scenario written in https://github.com/hmcts/ccd-case-management-web/pull/710/files.
> Maybe this is a good idea and I can see the benefit it'll bring(makes sure the test is in sync with the definition). We can keep this.
> The only concern I have is that it might result a bit confusing for who reads the scenario and is not a dev, because he won't understand why he has to pay attention to all the fields in the 'definition step' if they are then _apparently_ not used in the scenario (but we know they are used under the hood). I was confused at the beginning too.
> I wanted to raise this to stimulate a discussion

I agree with this. no point just putting it in all features/scenarios for the sake of it. if it brings no value to the test/feature then we should leave it out

Reply:
will update